### PR TITLE
Better generator name matching

### PR DESCRIPTION
**Pull request by pet-mit**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
In the CSE CRAC creator, creating an injection setpoint needs a generator in the network.
Currently, if the generator name in the CRAC (typically with wildcards) matches many buses in the network, farao cannot select the bus with a generator.

**What is the new behavior (if this is a feature change)?**
It is now possible to look through all the bus matches in order to select the only one having a generator.

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*
No, UcteBusHelper and GeneratorHelper are only used in CSE CRAC creator.

### CODE REVIEW

File: data/crac-creation/crac-creation-util/src/main/java/com/farao_community/farao/data/crac_creation/util/ucte/UcteBusHelper.java [hunk 1]
```diff
@@ -54,22 +56,19 @@ public UcteBusHelper(String nodeName, UcteNetworkAnalyzer ucteNetworkAnalyzer) {
             }
         }
 
-        // complex search with wildcard (either *, or incomplete ids)
```
Comment (baptsegu) [inline data/crac-creation/crac-creation-util/src/main/java/com/farao_community/farao/data/crac_creation/util/ucte/UcteBusHelper.java:57]: could you keep this comment

File: data/crac-creation/crac-creation-util/src/main/java/com/farao_community/farao/data/crac_creation/util/ucte/UcteMatchingResult.java [hunk 2]
```diff
@@ -25,17 +25,17 @@
     enum MatchStatus {
         OK(true),
```
Comment (baptsegu) [inline data/crac-creation/crac-creation-util/src/main/java/com/farao_community/farao/data/crac_creation/util/ucte/UcteMatchingResult.java:26]: UNIQUE_MATCH ? SINGLE_MATCH ?

File: data/crac-creation/crac-creation-util/src/main/java/com/farao_community/farao/data/crac_creation/util/ucte/UcteBusHelper.java (hunk 1 above)
Comment (pet-mit) [reply data/crac-creation/crac-creation-util/src/main/java/com/farao_community/farao/data/crac_creation/util/ucte/UcteBusHelper.java:57]: yes

File: data/crac-creation/crac-creation-util/src/main/java/com/farao_community/farao/data/crac_creation/util/ucte/UcteMatchingResult.java (hunk 2 above)
Comment (pet-mit) [reply data/crac-creation/crac-creation-util/src/main/java/com/farao_community/farao/data/crac_creation/util/ucte/UcteMatchingResult.java]: changed

**Review by baptsegu — APPROVED** [review]

**Pull request merged**